src/main/java/org/beanpod/switchboard/dto/LogDto.java

```diff
@@ -23,4 +26,16 @@
   private String level;
 
   private String serialNumber;
+
+  private StreamLog streamLog;
+
+  // for backward compatibility
+  public LogDto(
+      Long id, OffsetDateTime dateTime, String message, String level, String serialNumber) {
+    this.id = id;
+    this.dateTime = dateTime;
+    this.message = message;
+    this.level = level;
+    this.serialNumber = serialNumber;
```

Comment:
A way to pull this off without making a custom constructor is to mark all the properties except StreamLog as required, and add the `@RequiredArgsConstructor` annotation. Then it will generate 1 constructor with everything, and 1 constructor with only the required stuff (which is the one you are manually defining here).

Reply:
Thanks for letting me know about @RequiredArgsConstructor. I changed my code accordingly.